float32 magX
float32 magY
float32 magZ
